internal/testrunner/runners/pipeline/runner.go

```diff
@@ -256,6 +261,54 @@ func verifyDynamicFields(result *testResult, config *testConfig) error {
 	return nil
 }
 
+func convertNumericKeywordFields(result *testResult, config *testConfig) error {
```

Comment:
I wouldn't say that this should part of the pipeline test runner. I suppose a similar validation should be part of the system runner as well.

Please consider to put it in a common accessible place.